core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -123,41 +132,68 @@ export class PreferencesPageComponent {
   onSubjectInterestsSelectionChange(subjectInterests: string): void {
     this.alertsService.clearWarnings();
     this.subjectInterestsChangeAtLeastOnce = true;
-    this._saveDataItem('subject_interests', subjectInterests);
+    let data = this.PreferencesData();
+    data.subject_interests = subjectInterests;
+    this._saveDataItem(data);
   }
 
   savePreferredSiteLanguageCodes(preferredSiteLanguageCode: string): void {
     this.i18nLanguageCodeService.setI18nLanguageCode(preferredSiteLanguageCode);
-    this._saveDataItem(
-      'preferred_site_language_code', preferredSiteLanguageCode);
+    let data = this.PreferencesData();
+    data.preferred_site_language_code = preferredSiteLanguageCode;
+    this._saveDataItem(data);
   }
 
   savePreferredAudioLanguageCode(preferredAudioLanguageCode: string): void {
-    this._saveDataItem(
-      'preferred_audio_language_code', preferredAudioLanguageCode);
+    let data = this.PreferencesData();
+    data.preferred_audio_language_code = preferredAudioLanguageCode;
+    this._saveDataItem(data);
```

Comment:
Can you simplify this somehow? I think you can still do saveItem(which, to what) and then retrieve the `PreferencesData` inside the `saveIteme`.